modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/infinite_scroller/InfiniteScroller.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import PropTypes from 'prop-types';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+
+function InfiniteScroller({
+	children,
+	customLoader: CustomLoader,
+	maxHeight,
+	onBottomTouched,
+	scrollCompleted,
+}) {
+	const [scrollingAreaRendered, setScrollingAreaRendered] = useState(false);
+	const infiniteLoader = useRef(null);
+	const [infiniteLoaderRendered, setInfiniteLoaderRendered] = useState(false);
+	const scrollingArea = useRef(null);
+
+	const setScrollingArea = useCallback((node) => {
+		scrollingArea.current = node;
+		setScrollingAreaRendered(true);
+	}, []);
+
+	const setInfiniteLoader = useCallback((node) => {
+		infiniteLoader.current = node;
+		setInfiniteLoaderRendered(true);
+	}, []);
+
+	const setObserver = useCallback(() => {
+		if (
+			!scrollingArea.current ||
+			!infiniteLoader.current ||
+			!IntersectionObserver
+		) {
+			return;
+		}
+
+		const options = {
+			root: scrollingArea.current,
+			rootMargin: '0px',
+			threshold: 1.0,
+		};
+
+		const observer = new IntersectionObserver((entries) => {
+			if (entries[0].intersectionRatio === 1) {
+				onBottomTouched();
+			}
+		}, options);
```

Comment:
I don't know if you're planning on backporting this, but if you are, we should check that we're actually polyfilling `IntersectionObserver` in some way on IE (I don't recall if we are, and I don't know how easy — or even possible — it is to do so).